modules/install-upgrade/pages/upgrade-guide.adoc

```diff
@@ -0,0 +1,554 @@
++
+[source,bash]
+----
+helm get values > pulsar-backup-values.yaml
+----
++
+. To update the DataStax Pulsar Helm chart repository, run the following command:
++
+[source,bash]
+----
+helm repo update
+----
++
+. Open `helm/kaap-stack/values.yaml` and update the image tag to 3.1.0 (or the specific tag you wish to use).
++
+[source,yaml]
+----
+kaap:
+  enabled: true
+  cluster:
+    name: pulsar
+    create: true
+  spec:
+    global:
+      name: pulsar
+      image:
+        datastax/lunastreaming-all: 3.1_4.5
+----
++
+. Review and modify any other configuration parameters that may have changed between versions, such as resource limits, storage classes, and additional components. To modify other configurations, update `values.yaml` as needed. For example, to modify the broker's namespace shedding and splitting configurations, update the following fields:
++
+[source,yaml]
+----
+kaap:
+  enabled: true
+  cluster:
+    name: pulsar
+    create: true
+  spec:
+    global:
+      name: pulsar
+    broker:
+      replicas: 2
+      config:
+        loadBalancerNamespaceBundleSplitConditionHitCountThreshold: 1
+        loadBalancerSheddingConditionHitCountThreshold: 1e
+----
++
+. To upgrade your existing Pulsar installation, run the following Helm command.
+This command assumes the default `pulsar` namespace. If you are using a different namespace, replace `pulsar` with your namespace.
+The `--wait` flag ensures that Helm waits until all pods are ready before completing the upgrade.
++
+[source,bash]
+----
+helm upgrade pulsar --namespace pulsar --values current-values.yaml --wait
+----
++
+. Check the status of the pods to ensure they are running correctly:
++
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
+. Check the logs for any issues:
++
+[source,bash]
+----
+kubectl logs -n
+----
++
+. After upgrading, check if any additional configurations are required for new features in version 3.1. Adjust settings related to multi-tenancy, security, and observability as needed. Ensure all necessary configurations are in place and correct after the upgrade.
+. Test the functionality of your Pulsar cluster by sending messages and ensuring that consumers can read them without issues. Conduct functional tests to ensure that the upgrade did not impact existing applications and that new features work as expected.
+
+// known issues
+
+=== Kubernetes deployment using Helm chart
+
+The Helm chart for Luna Streaming is available in the https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[Helm chart sources] repository.
+
+Deploying Luna Streaming on Kubernetes using the DataStax Helm chart is another common method for running Pulsar in a cloud-native environment.
+
+. Back up your existing Pulsar data and configurations to prevent data loss.
+. To save your current Helm release configuration, run the following command:
++
+[source,bash]
+----
+helm get values > pulsar-backup-values.yaml
```

Comment:
Helm get values requires release name. Please update the command with below given command:
`helm get values <release-name> > pulsar-backup-values.yaml`